src/main/java/de/tum/in/www1/artemis/web/rest/ComplaintResource.java

```diff
@@ -246,6 +295,16 @@ public ComplaintResource(AuthorizationCheckService authCheckService, ExerciseSer
             complaints = complaintService.getAllComplaintsByExerciseIdAndTutorId(exerciseId, tutorId, isAtLeastInstructor);
         }
 
-        return ResponseEntity.ok(complaints);
+        return ResponseEntity.ok(getComplaintsByComplaintType(complaints, complaintType));
+    }
+
+    /**
+     * Filter out all complaints that are not of a specified type.
+     *
+     * @param complaints    list of complaints
+     * @param complaintType the type of complaints we want to get
+     */
+    private List<Complaint> getComplaintsByComplaintType(List<Complaint> complaints, ComplaintType complaintType) {
```

Comment:
This code seems duplicated in another class. Please double check and if this is the case, move it a common service class to avoid duplicated code!

Reply:
Didn't find any duplicate for this one.